### PR TITLE
If no flags are supplied, create empty flag object in application config.

### LINKED ISSUE #3475: Reloaded config after logging in causes flag error if none defined
Because the config is reloaded after logging in it is missing the
```
          if (!juju_config.flags) {
            juju_config.flags = {};
          }
```
in the index files. To remedy this we need to create a 'setter' for the config which will define this if not set.

### PR DESCRIPTION
**Pull request by hatched**
Fixes #3475

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

Load the GUI with no flags and with flags provided. It should work as expected.

File: jujugui/static/gui/src/app/init.js [hunk 1]
```diff
@@ -318,6 +318,19 @@ class GUIApp {
     }
   }
 
+  /**
+    Perform any processing or modification on the application config values.
+    @returns {Object} A processed applicationConfig
```
Comment (frankban) [inline jujugui/static/gui/src/app/init.js]: This needs a params line.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8491/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8492/

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge/2689

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**